default/scripting/techs/defense/mines.macros

```diff
@@ -271,3 +190,17 @@ EG_SYSTEM_MINES
                     empire = Target.Owner
             ]
 '''
+
+SYSTEM_CONTAINS_SOURCE_OR_OWNED_PLANET
+'''                ContainedBy And [
+                    System 
+                    Contains Or [
+                        Source
+                        And [
+                            Planet
+                            OwnedBy empire = Source.Owner
+                            OwnerHasTech name = ThisTech
```

Comment:
Why is this has tech condition needed? If this macro is being used in a tech script, then the source's owner will always have this tech. (And if it's used in something other than a tech, "ThisTech" doesn't make sense...)